### ISSUE TITLE
ZooKeeperProxy should not try to maintain canoncial versions of znodes

### ISSUE THREAD

**Issue by stephentu** [radlab/SCADS#26]
ZooKeeperProxy should not use the canonical map to pool znodes- since zookeeper watch semantics are "one time only", it is very difficult for the canonical map to be accurately updated when deletes from other clients are issued. this can cause many subtle bugs. since zookeeper's API allows us to explicit pass Watcher classes, we should instead use this feature for any watch callbacks we want to register

**Comment by marmbrus** [comment]
Yeah, you are right.  I don't think zookeeper had the ability to provide different watchers before (or I didn't know about it).  Given that that infrastructure already exists in the java client we should just use it.
